src/hooks/useGetSettlements.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useEffect, useState } from 'react'
+import { gql } from '@apollo/client'
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Network, UiError } from 'types'
+import { COW_SDK } from 'const'
+import { fetchSolversInfo } from 'utils/fetchSolversInfo'
+
+export const useGetSettlements = (
+  networkId: SupportedChainId = SupportedChainId.MAINNET,
+  initData: Settlement[],
+  solverAddress?: string,
+): GetSolverResult => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<UiError>()
+  const [settlements, setSettlements] = useState<Settlement[]>(initData)
+  const shouldRefetch = !initData.length
+
+  const fetchSettlements = useCallback(
+    async (network: Network): Promise<void> => {
+      setIsLoading(true)
+      setSettlements([])
+      try {
+        const response = await COW_SDK.cowSubgraphApi.runQuery<{
+          settlements: Settlement[]
+        }>(
+          solverAddress ? GET_SETTLEMENTS_BY_SOLVER_QUERY : GET_SETTLEMENTS_QUERY,
+          solverAddress ? { solver: solverAddress } : undefined,
```

Comment:
`solverAddress` should be lowercased, the subgraph doesn't work with checksummed addresses 

<img width="404" alt="image" src="https://user-images.githubusercontent.com/3328006/197284174-90704a9c-3e60-4742-893a-9fbe2f17d54b.png">